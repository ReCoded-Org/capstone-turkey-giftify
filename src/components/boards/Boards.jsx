import React, { useState } from "react";
import Cards from "../Cards/Cards";
import "./Board.css";
import { VscArrowRight } from "react-icons/vsc";

function Boards({ allItems, testSearchValue }) {
  const [selectedItems, setSelectedItems] = useState("");

  //  filter Cards by category
  const items = Object.keys(allItems)
    .filter((type) => type.includes(selectedItems))
    .map((item, index) => {
      return (
        <div key={index} className="cards">
          {allItems[item].map((info) => {
            return (
              <div key={index} className="oneCard">
                <img
                  className="itemImage"
                  src={info.image}
                  alt={info.itemName}
                />
                <p>
                  <b>{info.itemName}</b>
                </p>
                <a href="/link">
                  See Details <VscArrowRight />{" "}
                </a>
              </div>
            );
          })}
        </div>
      );
    });

  // Categories
  const category = Object.keys(allItems).map((categ, index) => {
    return (
      <div key={index} className="categories">
        <button
          id="oneCategory"
          className={categ}
          onClick={(event) => {
            event.preventDefault();
            setSelectedItems(categ);
          }}
          key={index}
        >
          {categ}
        </button>
        {categ === selectedItems ? items : null}
      </div>
    );
  });

  // Filter Cards by search
  const cardsBySearch = Object.keys(allItems)
    .filter((value) => {
      if (testSearchValue === "") {
        return null;
      } else if (value.toLowerCase().includes(testSearchValue.toLowerCase())) {
        return allItems[value];
      } else {
        return null;
      }
    })
    .map((item, index) => {
      return (
        <div key={index} className="cards">
          {allItems[item]
            .filter((value) => {
              if (testSearchValue === "") {
                return null;
              } else if (
                value.itemName
                  .toLowerCase()
                  .includes(testSearchValue.toLowerCase())
              ) {
                return value;
              } else if (
                value.condition
                  .toLowerCase()
                  .includes(testSearchValue.toLowerCase())
              ) {
                return value;
              } else {
                return value;
              }
            })
            .map((info) => {
              return (
                <div key={index} className="oneCard">
                  <img
                    className="itemImage"
                    src={info.image}
                    alt={info.itemName}
                  />
                  <p>
                    <b>{info.itemName}</b>
                  </p>
                  <a href="#link">
                    See Details <VscArrowRight />{" "}
                  </a>
                </div>
              );
            })}
        </div>
      );
    });

  return (
    <div>
      <Cards category={category} cardsBySearch={cardsBySearch} />
    </div>
  );
}

export default Boards;
