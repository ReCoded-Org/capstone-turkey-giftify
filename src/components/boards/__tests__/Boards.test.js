import React from "react";
import Boards from "./../Boards";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("matches Received", () => {
  const tree = renderer.create(<Boards allItems />).toJSON();
  expect(tree).toMatchSnapshot();
});
