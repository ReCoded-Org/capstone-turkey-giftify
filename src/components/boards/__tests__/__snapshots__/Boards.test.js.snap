// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Received 1`] = `
<div>
  <div />
</div>
`;

exports[`renders boards correctly 1`] = `
<div>
  <div />
</div>
`;
